spec/civildate.html

```diff
@@ -1,164 +1,566 @@
+      </p>
+      <emu-alg>
+        1. Let _date_ be the *this* value.
+        1. If Type(_date_) is not Object or _date_ does not have an [[InitializedTemporalDate]] internal slot, then
+          1. Throw a *TypeError* exception.
+        1. Let _otherDate_ be ? ToDate(_otherDate_).
+        1. Let _disambiguation_ be ? ToString(_disambiguation_).
+        1. If ! ValidateDate(_otherDate_.[[Year]], _otherDate_.[[Month]], _otherDate_.[[Day]]) is *false*, then
+          1. Throw a *TypeError* exception.
+        1. <mark>TODO: Throw if any time properties are present?</mark>
+        1. Let _greater_ be _date_.
+        1. Let _smaller_ be _otherDate_.
+        1. <mark>TODO: Swap order if needed.</mark>
+        1. Let _years_ be _greater_.[[Year]] - _smaller_.[[Year]].
+        1. Let _months_ be _greater_.[[Month]] - _smaller_.[[Month]].
+        1. Let _days_ be _greater_.[[Day]] - _smaller_.[[Day]].
+        1. If _months_ &lt; 0, then
+          1. Set _years_ to _years_ - 1.
+          1. Set _months_ to _months_ + 12.
+        1. If _days_ &lt; 0, then
+          1. Set _months_ to _months_ - 1.
+          1. Let _daysInMonth_ be DaysInMonth(_smaller_.[[Year]] + _years_, (_smaller_.[[Month]] + _months_) modulo 12)).
+          1. Set _days_ to _days_ + _daysInMonth_.
+        1. Return CreateDuration(_years_, _months_, _days_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause>
+      <h1>Temporal.Date.prototype.withTime ( _time_ )</h1>
+      <p>
+        The `withTime` method takes one argument _time_.
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _date_ be the *this* value.
+        1. If Type(_date_) is not Object or _date_ does not have an [[InitializedTemporalDate]] internal slot, then
+          1. Throw a *TypeError* exception.
+        1. Let _time_ be ? ToTimeLike(_time_).
+        1. <mark>TODO: Should this take a Temporal.Time instead?</mark>
+        1. Let _disambiguation_ be ? ToString(_disambiguation_).
+        1. If any of _time_.[[Hour]], _time_.[[Minute]], _time_.[[Second]], _time_.[[Millisecond]], _time_.[[Microsecond]], or _time_.[[Nanosecond]] is *undefined*, then
+          1. <mark>TODO: Merge into ToTimeLike?</mark>
+          1. Throw a *TypeError* exception.
+        1. Return CreateDateTime(_date_.[[Year]], _date_.[[Month]], _date_.[[Day]],
+          _time_.[[Hour]], _time_.[[Minute]], _time_.[[Second]],
+          _time_.[[Millisecond]], _time_.[[Microsecond]], _time_.[[Nanosecond]]).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause>
+      <h1>Temporal.Date.prototype.toString ( )</h1>
+      <p>
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _date_ be the *this* value.
+        1. If Type(_date_) is not Object or _date_ does not have an [[InitializedTemporalDate]] internal slot, then
+          1. Throw a *TypeError* exception.
+        1. Let _year_ be PadYear(_date_.[[Year]]).
```

Comment:
**Careful:** years smaller 1000 and larger 9999 are padded to 6 digits and always have a sign in front

 * 1 December 300 AD = +000300-12-01
 * 18 November 333 BC = -000333-11-18

but:

 * 18 November 1976 AD = 1976-18-11